managed/src/main/java/com/yugabyte/yw/common/KubernetesManager.java

```diff
@@ -52,94 +58,99 @@ public String getTimeout() {
-                                                         String universePrefix, boolean isMaster) {
+                                                         String namespace, boolean isMaster) {
     String serviceName = isMaster ? "yb-master-service" : "yb-tserver-service";
     List<String> commandList = ImmutableList.of("kubectl",  "get", "svc", serviceName,
-        "--namespace", universePrefix, "-o", "jsonpath=" + SERVICE_INFO_JSONPATH);
+        "--namespace", namespace, "-o", "jsonpath=" + SERVICE_INFO_JSONPATH);
     return execCommand(config, commandList);
   }
 
   public ShellResponse helmUpgrade(Map<String, String> config,
                                                        String universePrefix,
+                                                       String namespace,
                                                        String overridesFile) {
     String helmPackagePath = appConfig.getString("yb.helm.package");
     if (helmPackagePath == null || helmPackagePath.isEmpty()) {
       throw new RuntimeException("Helm Package path not provided.");
     }
     List<String> commandList = ImmutableList.of("helm",  "upgrade",  universePrefix,
-        helmPackagePath, "-f", overridesFile, "--namespace", universePrefix,
+        helmPackagePath, "-f", overridesFile, "--namespace", namespace,
         "--timeout", getTimeout(), "--wait");
     LOG.info(String.join(" ", commandList));
     return execCommand(config, commandList);
   }
 
   public ShellResponse updateNumNodes(Map<String, String> config,
-                                                          String universePrefix, int numNodes) {
-    List<String> commandList = ImmutableList.of("kubectl",  "--namespace", universePrefix, "scale",
+                                                          String namespace, int numNodes) {
+    List<String> commandList = ImmutableList.of("kubectl",  "--namespace", namespace, "scale",
         "statefulset", "yb-tserver", "--replicas=" + numNodes);
     return execCommand(config, commandList);
   }
 
   public ShellResponse helmDelete(Map<String, String> config,
-                                                      String universePrefix) {
+                                                      String universePrefix,
+                                                      String namespace) {
     List<String> commandList = ImmutableList.of("helm",  "delete", universePrefix,
-        "-n", universePrefix);
+        "-n", namespace);
     return execCommand(config, commandList);
   }
 
-  public void deleteStorage(Map<String, String> config, String universePrefix) {
+  public void deleteStorage(Map<String, String> config, String universePrefix, String namespace) {
     // Delete Master Volumes
     List<String> masterCommandList = ImmutableList.of("kubectl",  "delete", "pvc",
-        "--namespace", universePrefix, "-l", "app=yb-master");
+        "--namespace", namespace, "-l", "app=yb-master,release=" + universePrefix);
```

Comment:
nice! is the release label also present in helm 2?

Reply:
From the history of our charts repo, we have that label since the beginning, https://github.com/yugabyte/charts/commit/cb9f2f76b26c2196e86c7de7dd42783e286654c8